CDCgov/prime-reportstream
# Jessica/8862 default sender transform

This PR adds a default sender transform.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Make sure the tests still pass, especially the integration tests
2. In master, send a report using CA as the sender. Find the file that this created in the SFTP folder and note this for later. 
3. Switch to the PR's branch
5. Add schemaName: metadata/fhir_transforms/default-sender-transform to the organizations.yml file for the sender you are going to send to. 
6. Run the project, run reloadSettings, hit the same endpoint. 
7. View the file created in the sftp folder and make sure that it matches the file created earlier. 

## Changes
Created a default sender transform and moved the pieces into it that are needed.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #8862 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
N/A

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints? No
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints? No
- Does this change introduce new dependencies that need vetting? No
- Does this change require changes to our infrastructure? No
- Does logging contain sensitive data? No
- Does this PR include or remove any sensitive information itself? No

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
